lib/connections/service.go

```diff
@@ -370,15 +382,16 @@ func (s *Service) connect() {
 	}
 }
 
-func (s *Service) shouldLimit(addr net.Addr) bool {
-	if s.cfg.Options().LimitBandwidthInLan {
+func (s *Service) shouldLimit(addr net.Addr, options config.OptionsConfiguration) bool {
+	if options.LimitBandwidthInLan {
```

Comment:
I still think this if should be outside, as it's an early return which could be merged as part of the if condition without the mental and real stack overhead.